docs/theming/_index.md

```diff
@@ -25,60 +25,136 @@ To reference your theme, you have two options:
 - Using a URL, e.g. `"theme": "https://externalurl.example.com/theme-name/theme.json",`. To avoid CORS issues, please make sure that you host the URL on the same URL as your ownCloud web hosting.
 - For development and testing purposes, you can store your `theme.json` inside `packages/web-runtime/themes/{theme-name}/` and reference it in the `config.json`. However, this isn't recommended for production use since your changes may get lost when updating oCIS or the `web` app in OC10.
 
-**Hint:** If no theme is provided or the loading of your custom theme fails, the standard ownCloud theme will be loaded as a fallback. However, this doesn't stop you from correctly loading a theme that is wrongly formatted, so please read the instructions below carefully.
+**Hint:** If no theme is provided, the loading of your custom theme fails or the theme can't be parsed correctly, the standard ownCloud theme will be loaded as a fallback and an error with further information will be logged on the browser console.
 
 ## Configuring a theme
 
-Inside your `theme.json`, you can provide multiple themes as first-level objects. Currently, only the one called `"default"` gets applied when the frontend application is started. In the future, we'll provide functionality to dynamically switch between those themes.
+Inside your `theme.json`, there is a `common` key, which is explained in the next section, and a `clients` key: Here, you can find the available ownCloud clients - please note that the documentation below focuses on `web` and check the respective documentation for other clients for details on their themability.
 
-You can use the snippet below as a base for writing your own theme by replacing the strings and image file paths accordingly. Also, make sure to delete the comments from the file.
+The general top-level structure of a valid `theme.json` is outlined below:
+
+```json
+{
+  "common": {},
+  "clients": {
+    "android": {},
+    "desktop": {},
+    "ios": {},
+    "web": {}
+  }
+}
+```
+
+### Common section
+
+The `common` section provides a set of information that is designed to be available for all clients. It gets merged "down" to the final themes and aims to reduce duplication, but can be overwritten by more specific information inside both the clients' defaults and actual themes.
+
+The structure of a valid `comon` section is outlined below:
+
+```json
+"common": {
+  "name": "ownCloud",
+  "slogan": "ownCloud – A safe home for all your data",
+  "logo": "themes/owncloud/assets/logo.svg",
+  "urls": {
+    "accessDeniedHelp": "",
+    "imprint": "",
+    "privacy": ""
+  }
+}
+```
+
+- `name` specifies the publicly visible name for this oCIS instance
```

Comment:
My feeling is that the repetition of `for this oCIS instance` is a bit too much.